### ISSUE TITLE
Mutiny had to drop the following exception: (TIMEOUT,-1)

### ISSUE THREAD

**Issue by Nplu5** [quarkusio/quarkus#31045]
### Describe the bug

Mutiny throws an exception 30 seconds after calling `requestAnfForget` on the EventBus. I was experimenting with creating an asynchronous request response pattern in order to generate results file upon request. To make it simpler I started with an example of asynchronous greeting simulating the heavy lifting of the report generation with Thread.sleep().

I have a Resource with three Endpoints:

```kotlin
@Path("/greeting")
@ApplicationScoped
class GreetingResource(val bus: EventBus) {

    val statusUrl = "/greeting/status/{id}"

    @POST
    fun requestGreeting(): Response {
        val id = randomUUID()
        println("Request greeting")
        bus.requestAndForget<Message>("greeting", Message("Simon", id))
        println("Send 202 response")
        return Response
            .accepted()
            .header("Location", statusUrl.replace("{id}", id.toString()))
            .build()
    }

    @GET
    @Path("status/{id}")
    fun greetingStatus(@PathParam("id") id: UUID): Response {
        return if(MemoryDatabase.greetings.containsKey(id)){
            Response.seeOther(URI("/greeting/$id")).build()
        } else {
            Response.ok().entity("In progress").build()
        }
    }

    @GET
    @Path("/{id}")
    fun greeting(@PathParam("id") id: UUID): Response {
        return if (MemoryDatabase.greetings.containsKey(id)){
            val greeting = MemoryDatabase.greetings[id]
            MemoryDatabase.greetings.remove(id)
            Response.ok().entity(greeting).build()
        } else {
            Response.status(Response.Status.NOT_FOUND).entity("Greeting with id = $id was not found").build()
        }
    }
}
```

I request the greeting through posting to `requestGreeting()` and return `202`. I am not interested in the result of the consumer therefore using requestAndForget. The consumer of the Message then wait 32 seconds and saves the greeting in an "InMemoryDatabase"

```kotlin
data class Message(val name: String, val id: UUID)

@ApplicationScoped
class GreetingService {

    @ConsumeEvent(value= "greeting", blocking = true)
    fun generateGreeting(message: Message): String{
        println("Received Request to generate greeting")
        Thread.sleep(32000)
        MemoryDatabase.greetings[message.id] = "Hi ${message.name} I greet you asynchronously!"
        println("Generated greeting")
    }
}

object MemoryDatabase {
    val greetings = mutableMapOf<UUID, String>()
}
```
When checking the `greetingStatus` endpoint after 32 seconds I receive the message correctly but also receive an exception: 

```
2023-02-09 11:06:17,139 ERROR [io.qua.mut.run.MutinyInfrastructure] (vert.x-eventloop-thread-1) Mutiny had to drop the following exception: (TIMEOUT,-1) Timed out after waiting 30000(ms) for a reply. address: __vertx.reply.1, repliedAddress: greeting
	at io.vertx.core.eventbus.impl.ReplyHandler.handle(ReplyHandler.java:76)
	at io.vertx.core.eventbus.impl.ReplyHandler.handle(ReplyHandler.java:24)
	at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:932)
	at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:903)
	at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:55)
	at io.vertx.core.impl.DuplicatedContext.emit(DuplicatedContext.java:158)
	at io.vertx.core.impl.ContextInternal.emit(ContextInternal.java:194)
	at io.vertx.core.impl.VertxImpl$InternalTimerHandler.run(VertxImpl.java:921)
	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:153)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)

```

### Expected behavior

Exception should not be raised as answer from Consumer is not expected. 

### Actual behavior

Exception is thrown.

### How to Reproduce?

A Reproduction Repository can be found [here](https://github.com/Nplu5/Quarkus-async-request-response). The test `reproduceTimeoutException` simulates accessing the endpoint. The test passes but the log shows the exception.

### Output of `java -version`

OpenJDK Runtime Environment Temurin-11.0.17+8 (build 11.0.17+8)

### Quarkus version or git rev

2.16.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Build time:   2022-08-05 21:17:56 UTC Revision:     d1daa0cbf1a0103000b71484e1dbfe096e095918  Kotlin:       1.6.21 Groovy:       3.0.10 Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021 JVM:          11.0.15 (GraalVM Community 11.0.15+10-jvmci-22.1-b06) OS:           Windows 11 10.0 amd64

**Comment by Nplu5** [comment]
Here it is: https://github.com/Nplu5/Quarkus-async-request-response including as test `reproduceTimeoutException` to reproduce quickly. 

The repository was linked above under "How to Reproduce?" as well.

**Comment by cescoffier** [comment]
You can't block for so long (32s). The event bus gives up before that.

**Comment by cescoffier** [comment]
Also, request expect a response. You want to use send and not request

**Comment by Nplu5** [comment]
So for our real application we want to trigger the generation of a result file. As our data is very extensive it can take quite some time (10-20 minutes) to gather all the data and process it into a report. Thus, we wanted to leverage the asynchronous request-response pattern. Is the problem only ocurring because I am simulating the heavy work with Thread.sleep and thus being blocking? 

Regarding the second remark. in the documentation under Implementing fire and forget und der the [EventBus reference](https://quarkus.io/guides/reactive-event-bus#implementing-fire-and-forget-interactions) it gives an example for a method that does not return that's why I thought it would be possible. So is the documentation wrong or misleading?

**Comment by cescoffier** [comment]
For such kind of delay, you cannot use the event bus request/reply. It has a built-in timeout, and while you can increase it, I'm not sure it would be the right approach. 

Alternatively, you can use 2 adresses:

1. Send the request, includes an id,
2. Send the response when it's ready including the request id 

In term of code, the both side will have a @ConsumeEvent. Also, both side will use the event bus 'send' method (and not 'request').

About the documentation, it is showing an example valid when you use send or publish, but not request, as request implies a reply.

**Comment by Nplu5** [comment]
Thanks for clarification.  The `andForget` part of the method name led me to the conclusion that it does not need/care about a response.

**Comment by mh03r932** [comment]
I ran into the exact same issue with a use Case that is very Similar to Nplu5's.
I.e we have a Rest endpoint that must trigger a long running process but the http response should not wait on the result. I used the eventBus with `requestAndForget` to decouple the processing from the request. This worked but noticed the exception:
`(TIMEOUT,-1) Timed out after waiting 30000(ms)`

While I understand that this is more of a missunderstanding I still feel the documentation could be improved. 
On the [EventBus reference](https://quarkus.io/guides/reactive-event-bus#implementing-fire-and-forget-interactions) there is no mention of the EventBus#send method that seems to be the right tool for such a usecase. 

@cescoffier 
Maybe @Nplu5  example could be integrated into the documentation as an example on how to handle long running background tasks using the `send` method?
